lib/logstash/plugin_mixins/ecs_compatibility_support/target_check.rb

```diff
@@ -0,0 +1,46 @@
+# encoding: utf-8
+
+require_relative '../ecs_compatibility_support'
+
+module LogStash
+  module PluginMixins
+    module ECSCompatibilitySupport
+      # A target option check that can be included into any `LogStash::Plugin`.
+      #
+      # @see ECSCompatibilitySupport()
+      module TargetCheck
+        ##
+        # @api internal (use: `LogStash::Plugin::include`)
+        # @param base [Class]: a class that inherits `LogStash::Plugin`, typically one
+        #                      descending from one of the four plugin base classes
+        #                      (e.g., `LogStash::Inputs::Base`)
+        # @return [void]
+        def self.included(base)
+          fail(ArgumentError, "`#{base}` must inherit LogStash::Plugin") unless base < LogStash::Plugin
+          fail(ArgumentError, "`#{base}` must include #{ECSCompatibilitySupport}") unless base.method_defined?(:ecs_compatibility)
+        end
+
+        TARGET_NOT_SET_MESSAGE = ("ECS compatibility is enabled but no `target` option was specified, " +
+            "it is recommended to set the option to avoid potential schema conflicts (if your data is ECS compliant " +
+            "or non-conflicting feel free to ignore this message)").freeze
+
+        private
+
+        ##
+        # Logs an info message when ecs_compatibility is enabled but plugin has no `target` configuration specified.
+        # @note This method assumes a common plugin convention of using the target option.
+        # @return [nil] if ECS compatibility is disabled or no target option exists
+        # @return [true, false]
+        def check_target_set_in_ecs_mode
+          return if ecs_compatibility == :disabled || !respond_to?(:target)
```

Comment:
I'm not a fan of changing behaviour based on `respond_to?(:target)` at this stage -- if we have gotten this far and the plugin doesn't have a target, we have other problems and a silent change in behaviour is more likely to mask them than be helpful. Especially when other outside settings can result in the same behaviour.

Can we validate this earlier? Should it be an error condition to include `TargetCheck` on a plugin that doesn't have a `target` config option? In theory we can check it with `Module#included(base)`, but that would require this support module be included _after_ target is defined with the DSL. Is there another way?

Reply:
yes we can and there's no good reason to keep "working" if there's no target method ... will remove the check

Reply:
in the end I decided not to mess more due intercepting `target` config to keep things simple.
`register` will fail at runtime in ECS mode (any test with `ecs_compatibility` should reveal the issue)